src/pages/student/quizzes/QuizResult/Recent/index.js

```diff
@@ -1,21 +1,71 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import Card from 'react-bootstrap/Card';
 import { CgTimer } from 'react-icons/cg';
 import { PropTypes } from 'prop-types';
 
+import QuestionApi from '../../../../../api/Question';
+
 import style from './index.module.css';
 
-const Recent = (props) => {
+const Recent = ({ relatedQuiz, quizzes }) => {
+  const [questions, setQuestions] = useState(null);
+
+  useEffect(() => {
+    QuestionApi.getAll(relatedQuiz.id).then(({ data }) => {
+      setQuestions(data.data);
+    });
+  }, []);
+
+  const getTotalTimeLimit = () => {
+    if (questions != null) {
+      let total = 0;
+
+      for (let x = 0; x < questions.length; x++) {
+        total += questions[x].time_limit;
+      }
+
+      return total;
+    }
+  };
+
+  const getHighestScore = () => {
+    if (quizzes?.length > 0) {
+      const highestScore = quizzes?.reduce((prev, current) => {
+        return prev.score > current.score ? prev : current;
+      });
+
+      return highestScore.score;
+    }
+
+    return 0;
+  };
+
+  const getLatestScore = (quizId) => {
+    if (quizzes?.length > 0) {
+      let score = 0;
+
+      quizzes.forEach((quiz) => {
+        if (quiz.id === quizId) {
+          score = quiz.score;
+        }
+      });
+      return score;
+    }
+
+    return 0;
```

Comment:
This can be reduced by using `find()` method. Please find a way to reduce this further. Let me know if you have questions

```
myArray.find(x => x.id === quizId).score;
```

https://stackoverflow.com/questions/7364150/find-object-by-id-in-an-array-of-javascript-objects